examples/aws/cloudformation/oss.yaml

```diff
@@ -0,0 +1,910 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: Teleport OSS License
+Parameters:
+  KeyName:
+    ConstraintDescription: must be the name of an existing EC2 KeyPair.
+    Description: Name of an existing EC2 KeyPair to enable SSH access to the instances
+    Type: AWS::EC2::KeyPair::KeyName
+
+  InstanceType:
+    Description: Teleport EC2 instance type    
+    ConstraintDescription: must be a valid EC2 instance type.
+    Type: String    
+    Default: t2.micro
+    AllowedValues:
+    - t2.micro
+
+  DomainName:
+    ConstraintDescription: Teleport Web UI Domain Name
+    Description: Name of the domain name to use for Teleport
+    Type: String
+
+  DomainAdminEmail:
+    ConstraintDescription: Teleport Domain Admin email
+    Description: Name of Domain Name admin for Letsencrypt
+    Type: String
+
+  HostedZoneID:
+    ConstraintDescription: ID of the hosted zone with Teleport domain
+    Description: ID of the hosted zone of teleport domain
+    Type: String
+
+  # Any existing VPC used must have both DNS support and DNS hostnames enabled
+  VPC:
+    Description: Existing VPC ID to use (e.g. vpc-12312312)
+    Type: AWS::EC2::VPC::Id
+
+  # Public subnets to use for Teleport proxy
+  ProxySubnetA:
+    Description: ID of the first public proxy subnet (e.g. subnet-12312312)
+    Type: AWS::EC2::Subnet::Id
+
+  ProxySubnetB:
+    Description: ID of the second public proxy subnet (e.g. subnet-12312312)
+    Type: AWS::EC2::Subnet::Id
+
+  # Private subnets to use for Teleport auth
+  AuthSubnetA:
+    Description: ID of the first private auth subnet (e.g. subnet-12312312)
+    Type: AWS::EC2::Subnet::Id
+
+  AuthSubnetB:
+    Description: ID of the second private auth subnet (e.g. subnet-12312312)
+    Type: AWS::EC2::Subnet::Id
+
+  # Private subnets to use for Teleport nodes
+  NodeSubnetA:
+    Description: ID of the first private node subnet (e.g. subnet-12312312)
+    Type: AWS::EC2::Subnet::Id
+
+  NodeSubnetB:
+    Description: ID of the second private node subnet (e.g. subnet-12312312)
+    Type: AWS::EC2::Subnet::Id
+
+
+Mappings:
+  AWSInstanceType2Arch:
+    t2.micro: {Arch: HVM64}
+
+  AWSRegionArch2AMI:
+    # eu-west-1 = EU (Ireland)
+    # us-east-1 = US East (N. Virginia)
+    # us-east-2 = US East (Ohio)
+    # us-west-2 = US West (Oregon)
+    # All AMIs from AWS - gravitational-teleport-ami-oss-4.0.4
```

Comment:
I updated the OSS AMIs here - FYI we should always be using the `gravitational-teleport-ami-oss-*` format rather than the `marketplace` versions. The reason is that when we submit AMIs to AWS Marketplace, AWS likes to take ownership of those AMIs under their own account ID and those images aren't made public - they have their own special process for giving access to them for marketplace customers.

To combat this, we now build two AMIs for each region:

`gravitational-teleport-ami-oss-4.0.4` - our AMI, always owned by us, made public under our account, can be listed and used by anyone with a command like `aws ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-oss-4.0.4'` as per Terraform docs

`gravitational-teleport-marketplace-ami-oss-4.0.4` - AWS' AMI, will be moved under their account and made private when submitted to Marketplace